File: 04.BackendRTE/src/main/java/de/montigem/be/command/rte/general/GetAllCommandDTO.java
/* (c) https://github.com/MontiCore/monticore */

package de.montigem.be.command.rte.general;

public abstract class GetAllCommandDTO extends CommandDTO {
  protected int limit = -1;

  public GetAllCommandDTO limit(int limit) {
    this.limit = limit;
    return this;
  }
}

File: 03.FrontendRTE/src/app/shared/architecture/command/rte/getallcommand.dto.ts
/* (c) https://github.com/MontiCore/monticore */

import { JsonMember } from '@upe/typedjson';
import { CommandDTO } from "@shared/architecture/command/rte/command.dto";

export abstract class GetAllCommandDTO extends CommandDTO {
  @JsonMember({
    isRequired: false,
    name: 'limit',
    type: Number,
  }) public limit: number;

  protected constructor(typeName: string, limit?: number) {
    super(typeName);
    this.limit = limit;
  }
}
